console/src/app/modules/providers/provider-oauth/provider-oauth.component.ts

```diff
@@ -0,0 +1,298 @@
+import { Component, Injector, Type } from '@angular/core';
+import { AbstractControl, UntypedFormControl, UntypedFormGroup, Validators } from '@angular/forms';
+import { MatLegacyChipInputEvent as MatChipInputEvent } from '@angular/material/legacy-chips';
+import { ActivatedRoute, Router } from '@angular/router';
+import { take } from 'rxjs';
+import {
+  AddGenericOAuthProviderRequest as AdminAddGenericOAuthProviderRequest,
+  GetProviderByIDRequest as AdminGetProviderByIDRequest,
+  UpdateGenericOAuthProviderRequest as AdminUpdateGenericOAuthProviderRequest,
+} from 'src/app/proto/generated/zitadel/admin_pb';
+import { Options, Provider } from 'src/app/proto/generated/zitadel/idp_pb';
+import {
+  AddGenericOAuthProviderRequest as MgmtAddGenericOAuthProviderRequest,
+  GetProviderByIDRequest as MgmtGetProviderByIDRequest,
+  UpdateGenericOAuthProviderRequest as MgmtUpdateGenericOAuthProviderRequest,
+} from 'src/app/proto/generated/zitadel/management_pb';
+import { AdminService } from 'src/app/services/admin.service';
+import { Breadcrumb, BreadcrumbService, BreadcrumbType } from 'src/app/services/breadcrumb.service';
+import { ManagementService } from 'src/app/services/mgmt.service';
+import { ToastService } from 'src/app/services/toast.service';
+
+import { PolicyComponentServiceType } from '../../policies/policy-component-types.enum';
+
+@Component({
+  selector: 'cnsl-provider-oauth',
+  templateUrl: './provider-oauth.component.html',
+  styleUrls: ['./provider-oauth.component.scss'],
+})
+export class ProviderOAuthComponent {
+  public showOptional: boolean = false;
+  public options: Options = new Options();
+
+  public id: string | null = '';
+  public updateClientSecret: boolean = false;
+  public serviceType: PolicyComponentServiceType = PolicyComponentServiceType.MGMT;
+  private service!: ManagementService | AdminService;
+  public readonly separatorKeysCodes: number[] = [ENTER, COMMA, SPACE];
+  public form!: UntypedFormGroup;
+
+  public loading: boolean = false;
+
+  public provider?: Provider.AsObject;
+
+  constructor(
+    private router: Router,
+    private route: ActivatedRoute,
+    private toast: ToastService,
+    private injector: Injector,
+    private _location: Location,
+    breadcrumbService: BreadcrumbService,
+  ) {
+    this.form = new UntypedFormGroup({
+      name: new UntypedFormControl('', [Validators.required]),
+      clientId: new UntypedFormControl('', [Validators.required]),
+      clientSecret: new UntypedFormControl('', [Validators.required]),
+      authorizationEndpoint: new UntypedFormControl('', [Validators.required]),
+      tokenEndpoint: new UntypedFormControl('', [Validators.required]),
+      userEndpoint: new UntypedFormControl('', [Validators.required]),
+      idAttribute: new UntypedFormControl('', [Validators.required]),
+      scopesList: new UntypedFormControl(['openid', 'profile', 'email'], []),
+    });
+
+    this.route.data.pipe(take(1)).subscribe((data) => {
```

Comment:
Shouldn't this be in the pages directory?
If not, shouldn't it be more reactive than `take(1)`?
Can you explain me your thought about that in a meet?

Reply:
no as it is reused in two pages.. 👍 , the console does't change the id from the code so `take(1)` is enough